core/jobs/batch_jobs/test_gcs_io_job_test.py

```diff
@@ -0,0 +1,120 @@
+# coding: utf-8
+#
+# Copyright 2022 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for gcs_io."""
+
+from __future__ import annotations
+
+from core.jobs import job_test_utils
+from core.jobs.batch_jobs import test_gcs_io_job
+from core.jobs.types import job_run_result
+from core.platform import models
+
+from typing import List
+
+MYPY = False
+if MYPY:  # pragma: no cover
+    from mypy_imports import app_identity_services
+    from mypy_imports import storage_services
+
+storage_services = models.Registry.import_storage_services()
+app_identity_services = models.Registry.import_app_identity_services()
+
+
+class TestGCSIoWriteJobTests(job_test_utils.JobTestBase):
+
+    JOB_CLASS = test_gcs_io_job.TestGCSIoWriteJob
+
+    def test_to_fetch_exp_filename(self) -> None:
+        self.assert_job_output_is([
+            job_run_result.JobRunResult(stdout='TOTAL FILES WRITTEN SUCCESS: 2')
+        ])
+
+
+def write_files_to_gcs() -> List[str]:
+    """Write dummy files to GCS."""
+    bucket = app_identity_services.get_gcs_resource_bucket_name()
+    fileobjects = [
+        {
+            'filepath': 'dummy_folder/dummy_subfolder/dummy_file_1',
+            'data': b'testing_1'
+        },
+        {
+            'filepath': 'dummy_folder/dummy_subfolder/dummy_file_2',
+            'data': b'testing_2'
+        }
+    ]
+
+    filepath_0 = fileobjects[0]['filepath']
+    assert isinstance(filepath_0, str)
+    data_0 = fileobjects[0]['data']
+    assert isinstance(data_0, bytes)
+    filepath_1 = fileobjects[1]['filepath']
+    assert isinstance(filepath_1, str)
+    data_1 = fileobjects[1]['data']
+    assert isinstance(data_1, bytes)
```

Comment:
You don't need so many asserts if you correctly set the type for `fileobjects` (`FileObjectDict`)

Reply:
Done